### PR TITLE
Add Snapcraft build metadata

### PR DESCRIPTION
**Pull request by RAOF**
Snaps (https://snapcraft.io/) are a Linux application distribution format.

This adds the requisite metadata to build a GPMDP snap from the repository.
The major features work - MPRIS integration, desktop indicators/systray, music hotkeys and so on.

The desktop file icon doesn't work, and you will get an all-white window unless you manually connect the “process-control” snap interface.

With this metadata in the repository we can get Launchpad (http://launchpad.net/) to do daily builds of the GPMDP snap and publish them to “edge” channel of the snap store.

### CODE REVIEW

File: snap/gui/google-play-music-desktop-player.desktop [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
```
Comment (MarshallOfSound) [inline snap/gui/google-play-music-desktop-player.desktop]: Can this be moved to `vendor` and somehow indicate that it is related to `snap`.

Something like `vendor/snap/google-play-music-desktop-player.desktop`

File: .gitignore [hunk 2]
```diff
@@ -9,5 +9,13 @@ vendor/*.msi
 electron.xml
 spectron.xml
 coverage.zip
-uwp/
```
Comment (MarshallOfSound) [inline .gitignore]: This still needs to be here

File: snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+description: |
+  This Google Play Music player provides desktop integration - with music
+  hotkeys and the sound indicator via an MPRIS interface, the abiliy to use
+  custom colours and load custom themes, a remote-control interface, and
+  lower resource consumption than a full-blown browser tab.
+  .
+  This is *not* an official Google project.
+grade: stable
+confinement: strict
+icon: snap/gui/google-play-music-desktop-player.png
```
Comment (MarshallOfSound) [inline snapcraft.yaml]: Can this point to the PNG generated as part of the build phase?

Comment (RAOF) [inline snapcraft.yaml]: Sadly, no. The `snapcraft` tool checks for the presence of the `icon:` file at the beginning of the build.

Oh. But it accepts SVG, so it *could* point to `src/assets/icons/svg/vector_logo.svg`. Is there any reason not to do that?

**Comment by RAOF** [comment]
To get this building and deploying I can add a Launchpad import of the github repository; that will pull the branch four times a day and build + publish the snap (to the “edge” channel) if there are any changes.

If you'd prefer to do a build per commit, the `snapcraft enable-ci travis` command is supposed to be able to do the necessary changes to add a snapcraft build to Travis.

Launchpad has the advantage that it won't slow down CI and we'll be able to extend this (after a couple of snapcraft bugs have been fixed) to build packages for arm, arm64, i386, and (for the truly weird) PowerPC. It's got the disadvantage that changes which break the snap build won't be noticed by CI.

Once the builds are in the edge channel, you can promote them to the beta/candidate/stable channels from the developer page on the snap store (or, at least, you'll be able to once I've set that up 😏). There are other possibilities, like setting the git-tag to build from in the `snapcraft.yaml` file, or manually building locally and uploading, if you'd prefer.

**Comment by MarshallOfSound** [comment]
If you delete the PNG and rebase we can try get this going for `4.2.0`

**Comment by Ads20000** [comment]
Ah sorry I filed an issue before checking closed issues/PRs, silly me, @RAOF how's setting up the developer page on the Snap store for @MarshallOfSound to use? They're adding a feature which will allow you to add additional contributors to a Snap which I think might help, but as far as I know that's still in the works...

**Pull request merged**